### ISSUE TITLE
FR: add argument to disable linting of `(` for auto-printing  in implicit_assignment_linter

### ISSUE THREAD

**Issue by TimTaylor** [jimhester/lintr#2916]
Would it be possible to gain a parameter in the `implicit_assignment_linter` to allow for implicit printing? At present, I see no way to disable this, e.g

``` r
lintr::lint(
    text = "(a <- 1)", 
    linters = lintr::implicit_assignment_linter()
)
#> <text>:1:2: warning: [implicit_assignment_linter] Call print() explicitly instead of relying on implicit printing behavior via '('.
#> (a <- 1)
#>  ^~~~~~

lintr::lint(
    text = "(a <- 1)", 
    linters = lintr::implicit_assignment_linter(except = "`(`")
)
#> <text>:1:2: warning: [implicit_assignment_linter] Call print() explicitly instead of relying on implicit printing behavior via '('.
#> (a <- 1)
#>  ^~~~~~
```

**Comment by MichaelChirico** [comment]
That makes sense to me, do you want to have a go at a PR?
